VitruvioHost/Plugins/Vitruvio/Source/Vitruvio/Private/InitialShape.cpp

```diff
@@ -241,8 +241,91 @@ TArray<FInitialShapeFace> CreateDefaultInitialFaces()
 	return InitialFaces;
 }
 
+bool IsCongruentToDefaultInitialShape(const TArray<FInitialShapeFace>& InitialFaces)
+{
+	const TArray<FInitialShapeFace> DefaultInitialFaces = CreateDefaultInitialFaces();
+	check(DefaultInitialFaces.Num() == 1);
+	const TArray<FVector> DefaultVertices = DefaultInitialFaces[0].Vertices;
+	check(DefaultVertices.Num() == 4);
+
+	if (InitialFaces.Num() == DefaultInitialFaces.Num())
+	{
+		const TArray<FVector> Vertices = InitialFaces[0].Vertices;
+
+		if(Vertices.Num() == DefaultVertices.Num())
+		{
+			const FVector FirstVertex = DefaultVertices[0];
+			int32 InitialIndexOffset;
+			if(Vertices.Find(FirstVertex,InitialIndexOffset))

```

Comment:
How are we dealing with slight float differences? Are expecting exact equality, or is UE4's comparison dealing with tolerances?

Reply:
This code should mainly handle if a user switches between spline and staticmesh on the default mesh. Therefore they should actually be equal, but we can change it for robustness. Unreal has a .Equals function that uses a threshold.